config/v1/types_cluster_version.go

```diff
@@ -106,6 +111,48 @@ type ClusterVersionStatus struct {
 	AvailableUpdates []Update `json:"availableUpdates"`
 }
 
+// UpdateState is a constant representing whether an update was successfully
+// applied to the cluster or not.
+type UpdateState string
+
+const (
+	// CompletedUpdate indicates an update was successfully applied
+	// to the cluster (all resource updates were successful).
+	CompletedUpdate UpdateState = "Completed"
+	// PartialUpdate indicates an update was never completely applied
+	// or is currently being applied.
+	PartialUpdate UpdateState = "Partial"
+)
+
+// UpdateHistory is a single attempted update to the cluster.
+type UpdateHistory struct {
```

Comment:
`UpdateHistory` is `Update` + `state`. so should this embedded `Update` object ie extend rather than adding version and payload fields?

Reply:
I would like to have different Godoc on Update and UpdateHistory for those fields (I realized I need to elide the comment about availableUpdates)

Reply:
Updated with different godoc (demonstrating why)